ga4gh/frontend.py

```diff
@@ -233,6 +241,48 @@ def configure(configFile=None, baseConfig="ProductionConfig", extraConfig={}):
     theBackend.setDefaultPageSize(app.config["DEFAULT_PAGE_SIZE"])
     theBackend.setMaxResponseLength(app.config["MAX_RESPONSE_LENGTH"])
     app.backend = theBackend
+    app.secret_key = os.urandom(SECRET_KEY_LENGTH)
+    app.oidcClient = None
+    app.tokenMap = None
+    app.myPort = port
+    if "OIDC_PROVIDER" in app.config:
+        app.oidcClient = oic.oic.Client()
+        app.tokenMap = {}
+        try:
+            app.oidcClient.provider_config(app.config['OIDC_PROVIDER'])
+        except requests.exceptions.ConnectionError:
+            configResponse = message.ProviderConfigurationResponse(
+                issuer=app.config['OIDC_PROVIDER'],
+                authorization_endpoint=app.config['OIDC_AUTHZ_ENDPOINT'],
+                token_endpoint=app.config['OIDC_TOKEN_ENDPOINT'],
+                revocation_endpoint=app.config['OIDC_TOKEN_REV_ENDPOINT'])
+            app.oidcClient.handle_provider_config(configResponse,
+                                                  app.config['OIDC_PROVIDER'])
+
+        # The redirect URI comes from the configuration.
+        # If we are testing, then we allow the automatic creation of a
+        # redirect uri if none is configured
+        redirectUri = app.config.get('OIDC_REDIRECT_URI')
+        if redirectUri is None and 'TESTING' in app.config:
+            redirectUri = 'https://{0}:{1}/oauth2callback'.format(
+                socket.gethostname(), app.myPort)
+        app.oidcClient.redirect_uris = [redirectUri]
+        if redirectUri is []:
```

Comment:
What are we testing for here? We should raise a more informative error in any case.